### PR TITLE
[#140] Creating Component for search filters/bar

### PR DESCRIPTION
**Pull request by Jbcampbe**
Between the search and scan page we had complete duplicated code which was hard to maintain and test. This PR...

1) extracts those common things and puts them in a `search-header` component. 

2) created a `select-dropdown` component to make creating a dropdown easy anywhere in the app.

3) adds tests for the both of those components (you can run them by using `ember test`).

Hopefully if I did everything correctly the ui should look and behave exactly the same. I hope to continue to go through and do more of these cleanup and write tests for the whole app.

Sorry I know it is a lot of changes ; )

@amenders I took the liberty of making css class name and css changes. Hope thats ok : )

### CODE REVIEW

File: app/mixins/search-driver.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import Ember from 'ember';
```
Comment (ekraihan) [inline app/mixins/search-driver.js]: just for learning's sake, wanna give a brief explanation of 'mixins' ? :)

**Comment by Jbcampbe** [comment]
In a mixin you can define properties and functions. Then when you "mix it in" to an ember object (like a controller) that object gets those properties and that function.

This is my first time using one and I just discovered that if you use the mixin on multiple objects the data is shared... so that is why the query objects aren't in there. I think there is a way to disable the data sharing so I'm gonna do that and include it in the PR if I can.

_____________

**Comment by Jbcampbe** [comment]
Just moved the data to the mixin.

File: app/mixins/search-driver.js (hunk 1 above)
Comment (Jbcampbe) [inline app/mixins/search-driver.js]: I sent an email reply. TLDR its kinda like creating a base class that you can "inherit" data and methods from.

**Comment by Jbcampbe** [comment]
@ekraihan objections to this merging?

**Comment by Jbcampbe** [comment]
@ekraihan I just fixed a couple conflicts created by merging my acceptance testing PR. Any concerns about a merge?

**Pull request merged**